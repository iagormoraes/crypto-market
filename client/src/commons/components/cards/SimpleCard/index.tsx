import { HTMLAttributes } from 'react';
import classnames from 'classnames';

interface Props extends HTMLAttributes<HTMLDivElement> {
  title: string;
  children?: any;
}

export default function SimpleCard({
  title,
  children,
  className,
  ...props
}: Props) {
  return (
    <div
      className={classnames(
        'rounded shadow-xl w-full md:w-48 overflow-hidden',
        className,
      )}
      {...props}
    >
      <div>
        <h3 className="text-center text-white font-bold p-2 bg-yellow-500">
          {title}
        </h3>
        <div className="p-2">{children}</div>
      </div>
    </div>
  );
}
