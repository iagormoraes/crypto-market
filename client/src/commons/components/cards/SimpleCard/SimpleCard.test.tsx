import { getByText, render, screen } from '@testing-library/react';

import SimpleCard from '.';

describe('<SimpleCard/>', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <SimpleCard title="title">children</SimpleCard>,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders title and description', async () => {
    const props = {
      title: 'Title',
      children: <p>children</p>,
    };

    render(<SimpleCard title={props.title}>{props.children}</SimpleCard>);

    expect(screen.getByText(props.title)).toBeInTheDocument();
    expect(screen.getByText('children')).toBeInTheDocument();
  });
});
