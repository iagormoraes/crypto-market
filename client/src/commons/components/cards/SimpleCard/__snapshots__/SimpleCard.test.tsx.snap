// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleCard/> matches snapshot 1`] = `
<div>
  <div
    class="rounded shadow-xl w-full md:w-48 overflow-hidden"
  >
    <div>
      <h3
        class="text-center text-white font-bold p-2 bg-yellow-500"
      >
        title
      </h3>
      <div
        class="p-2"
      >
        children
      </div>
    </div>
  </div>
</div>
`;
